### DISCUSSION TITLE
Updating windows setup documentation

### DISCUSSION THREAD

**Discussion by derekthecool** [Frederick888/external-editor-revived#78] [General]
Thanks for this great tool! Amazing!

I've found by my testing and setup today that use mintty or other UNIX like windows shells is unneeded.

I was able to get this application setup and working by downloading the external editor exe file and running it from powershell (pwsh to be precise). I skipped all the  `chmod` stuff used by mintty.

Then in my config settings from thunder bird I selected custom setup using the following values:

shell: `pwsh` (full path not given)
command template : `alacritty -e nvim "/path/to/temp.eml"`

**Comment by Frederick888** [comment]
That's great to know! I added a link to this discussion in [wiki](https://github.com/Frederick888/external-editor-revived/wiki/Windows#powershell).

Out of curiosity:

- Does Neovim run directly in PowerShell?
- Did you manually add Alacritty to your search path, or their installer took care of it automatically?
- If Neovim does not run in PowerShell, are there any other Windows-compatible terminal emulators that came into your mind?

**Reply by derekthecool** [reply]
* Yes, neovim can run directly in powershell.exe (version 5.1 and lower) or pwsh (version 6.0+).
* I use [scoop](https://scoop.sh/) to install programs such as neovim, Alacritty, and thunder bird with commands like this
```powershell
scoop install neovim alacritty thunderbird
```
Scoop handles adds entries to the system path for these programs.

Here is my full config command

```txt
# Editor
custom

# Shell
pwsh

# Command template
alacritty --title "Email" -e nvim "/path/to/temp.eml"
```

**Comment by ikenichiro** [comment]
Thank you so much for reviving one of the best extension for Thunderbird!
I'm using windows as also, I would like to add some additional information that might be helpful for windows users.
Or, for those using Visual Studio Code (VSCode) should be able to reuse the command template on any OS.

- My Configuration
  - I also use scoop to install program, but for me, I use Thunderbird as windows native app (=Thunderbird not installed through scoop).
  - I wanted to use vscode + neovim (using VSCode Neovim extension)
` scoop install vscode neovim`

I tested my full config with the following settings, which are
```
# Editor
custom

# Shell
sh

# Command template
code -n -w "/path/to/temp.eml"
```

In addition, I setup a temporary directory inside the default temporary directory so I can trust the working directory for vscode. e.g. `C:\Users\MYUSERNAME\AppData\Local\Temp\ExternalEditorRevived` (replace MYUSENAME with your default username on Windows)


- Some additional information
    - Having git installed through scoop (which is mandatory using scoop), you won't need cygwin, since you can use git bash instead.
    - All the environmental path are managed by scoop, and `sh` is included in the path so you should be able to use the predefined default configurations for this extension (instead of using custom configurations)
      - I was able to use `gVim` from the drop down list of Editor, having gvim also installed by scoop `scoop install gvim`

**Comment by chrnz008** [comment]
can i  know what is temp.eml i only used a json file for native messaging host

**Comment by Zevan770** [comment]
I've made a scoop manifest for the message host:
https://github.com/Zevan770/scoop_zev/blob/077a9a2d1e6c70416a0c8994d2ec709d75fcfa6f/bucket/external-editor-revived.json

use it by:

```pwsh
scoop bucket add zev https://github.com/Zevan770/scoop_zev
scoop install external-editor-revived
```

After installing the .exe executable, `post_install` script will automatically create a message host manifest file at scoop persist directory and create the registry accordingly.
